README.md

```diff
@@ -1,48 +1,114 @@
+
+Nixpkgs and NixOS are built and tested by our continuous integration
+system, [Hydra](https://hydra.nixos.org/).
 
-* [NixOS installation instructions](https://nixos.org/nixos/manual/#ch-installation)
-* [Documentation (Nix Expression Language chapter)](https://nixos.org/nix/manual/#ch-expression-language)
-* [Manual (How to write packages for Nix)](https://nixos.org/nixpkgs/manual/)
-* [Manual (NixOS)](https://nixos.org/nixos/manual/)
-* [Community maintained wiki](https://nixos.wiki/)
 * [Continuous package builds for unstable/master](https://hydra.nixos.org/jobset/nixos/trunk-combined)
-* [Continuous package builds for 19.03 release](https://hydra.nixos.org/jobset/nixos/release-19.03)
+* [Continuous package builds for the NixOS 19.03 release](https://hydra.nixos.org/jobset/nixos/release-19.03)
 * [Tests for unstable/master](https://hydra.nixos.org/job/nixos/trunk-combined/tested#tabs-constituents)
-* [Tests for 19.03 release](https://hydra.nixos.org/job/nixos/release-19.03/tested#tabs-constituents)
+* [Tests for the NixOS 19.03 release](https://hydra.nixos.org/job/nixos/release-19.03/tested#tabs-constituents)
 
-Communication:
+Artifacts successfully built with Hydra are published to cache at
+https://cache.nixos.org/. When successful build and test criteria are
+met, the Nixpkgs expressions are distributed via [Nix
+channels](https://nixos.org/nix/manual/#sec-channels). The channels
+are provided via a read-only mirror of the Nixpkgs repository called
+[nixpkgs-channels](https://github.com/NixOS/nixpkgs-channels).
 
-* [Discourse Forum](https://discourse.nixos.org/)
-* [IRC - #nixos on freenode.net](irc://irc.freenode.net/#nixos)
+# Contributing
+
+Nixpkgs is among the most active projects on GitHub. While thousands
+of open issues and pull requests might seem a lot at first, it helps
+consider it in the context of the scope of the project. Nixpkgs
+describes how to build over 40,000 pieces of software and implements a
+Linux distribution. The [GitHub Insights](https://github.com/NixOS/nixpkgs/pulse)
+page gives a sense of the project activity.
+
+Community contributions are always welcome through GitHub Issues and
+Pull Requests. When pull requests are made, our tooling automation bot,
+[OfBorg](https://github.com/NixOS/ofborg) will perform various checks
+to help ensure expression quality.
+
+The *Nixpkgs maintainers* are people who have assigned themselves to
+maintain specific individual packages. We encourage people who care
+about a package to assign themselves as a maintainer. When a pull
+request is made against a package, OfBorg will notify the appropriate
+maintainer(s). The *Nixpkgs committers* are people who have been given
+permission to merge.
+
+Most contributions are based on and merged into these branches:
+
+* `master` is the main branch where all small contributions go
+* `staging` is branched from master, and mass-rebuilds (changes
+  affecting over 100 packages) and other large contributions go to this
```

Comment:
I've heard that staging is actually suitable for much larger rebuilds. 501+ I think.

Reply:
It would be nice to better define this somewhere. Maybe I need to take this out and make an RFC. As a data point, I settled on making r-ryantm PR to staging if it results in 100 or more package rebuilds.

Reply:
I would prefer if staging was targeted whenever ofborg is not able to build the PR. That way master could always be green.

Reply:
On second thought, I'm probably being too rigid about this and can write in in a more flexible way. I'll take another stab at it.

Reply:
A qualifier, a word, an adjective, should be used instead of using numbers in the README for the description for branches. This is not a guide for contributors, but an orientation document for new users.

Something like "... mass-rebuilds (changes having a high rebuild cost) and other large ..." would stay relevant whatever the perceived limit for staging is.

It is especially true when one considers that all derivations are not equal. Some derivations amount to "download → extract → copy" and some are heavy compilations! I wouldn't hesitate to send a 1000 trivial rebuild to master, while I would think about sending a 100 heavy builds to staging depending.

Reply:
There is no hard line as it depends on what is being rebuild and the state of Hydra. As a guideline, put 500+ in staging. If we put smaller-rebuild commits in there, then too many PR's will get blocked by required changes in staging, whereas if we only do very large rebuilds, trunk will be rebuilding all the time blocking staging.

Reply:
I think I addressed this issue; please check it out.

Reply:
Good way to address this too. Not being quantatative in the terms helped.